web/src/hooks/setting/useSettingState.js

```diff
@@ -0,0 +1,84 @@
+import React, {useEffect, useReducer, useState} from "react"
+import useGetSetting from "./useGetSetting"
+import {saveSettings} from "../../api"
+import {ACTIONS, reducer} from "./reducer";
+import {INITIAL_STATE} from "./default";
+
+const useSettingState = () => {
+  const [state, dispatch] = useReducer(reducer, INITIAL_STATE)
+
+  const [isFirst, setIsFirst] = useState(true);
+  const {isLoading, data, error} = useGetSetting()
+
+  useEffect(() => {
+    if (!isLoading) {
+      dispatch({type: ACTIONS.INIT_STATE, payload: data})
+    }
+  }, [data, error])
+
+  useEffect(() => {
+    if (!isFirst && !isLoading) {
+      saveSettings(state).then()
+    } else {
+      setIsFirst(false)
+    }
+  }, [
+    state,
+  ])
```

Comment:
```suggestion
  useEffect(() => {
    if (isFirst) {
      setIsFirst(false)
      return
    }
  
    saveSettings(state)
  }, [
    state,
  ])
```

На самом деле можно пойти дальше и использовать вместо `setIsFirst` рефы:
```ts
const isFirstRef = useRef(true)

useEffect(() => {
  if (isFirstRef.current) {
     isFirstRef.current = false
     return
   }
  
   saveSettings(state)
  }, [
    state,
  ])
```

useState вызывает ре-рендр, а тебе он здесь по идее не нужен, поэтому можно использовать ref для хранения этого флага. Но это не крит